packages/tooltip/src/vaadin-tooltip-mixin.js

```diff
@@ -447,6 +447,7 @@ export const TooltipMixin = (superClass) =>
       this._isConnected = true;
 
       document.body.addEventListener('vaadin-overlay-open', this.__onOverlayOpen);
+      document.addEventListener('mousedown', this.__onMouseDown);
```

Comment:
I wonder if we should rather fix this by implementing `bringToFront` for Notification also. Dialogs / other overlays can get opened by other means than clicks / keydowns also

Reply:
Yes, that's a valid concern but I think the fix is still relevant since the problem here is that `hideDelay` defers closing and `zIndex` is set unexpectedly (we have no way to detect that overlay is about to be closed).

Reply:
In that case, notification would dynamically get an even higher `zIndex` via `bringToFront`. Currently, it has a static [value](https://github.com/vaadin/web-components/blob/0f883924dc1bc039c67d5b78091290a388ec33b6/packages/notification/src/vaadin-notification-styles.js#L11) of 1000 which has been assumed to make it cover all other overlays.

Reply:
Good point. I can make a PR to implement `bringToFront()` (maybe as a protected API for now).

Reply:
Tooltip also uses `bringToFront` so it wouldn't be an issue

Reply:
Actually if we get the `vaadin-notification-container` to use `OverlayStackMixin` then setting higher z-index on the tooltip overlay might be unnecessary since it was specifically added to address that https://github.com/vaadin/web-components/pull/6074

Reply:
Created #8198 which adds notification container to the overlay stack using corresponding mixin.